File: flutter/library/lib/src/tsp_text_link.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';

class TspTextLink extends StatelessWidget {
  const TspTextLink({super.key, required this.text, this.inverse = false, this.theme = StarPlanetTheme.sky, this.onTap});
  final String text;
  final bool inverse;
  final StarPlanetTheme theme;
  final VoidCallback? onTap;

  @override
  Widget build(BuildContext context) => GestureDetector(
        onTap: onTap,
        child: Text(text, style: TextStyle(color: inverse ? Colors.white : theme.brandPrimary, fontWeight: FontWeight.w800)),
      );
}

File: ios-swiftui/library/Sources/PlanetComponents/TspTextLink.swift
import SwiftUI

public struct TspTextLink: View {
    let text: String
    let inverse: Bool
    let theme: StarPlanetTheme
    public init(_ text: String, inverse: Bool = false, theme: StarPlanetTheme = .sky) {
        self.text = text
        self.inverse = inverse
        self.theme = theme
    }
    public var body: some View {
        Text(text).fontWeight(.bold).foregroundColor(inverse ? .white : theme.brandPrimary)
    }
}
